gax-httpjson/src/main/java/com/google/api/gax/httpjson/ManagedHttpJsonChannel.java

```diff
@@ -39,17 +39,26 @@
 import com.google.api.gax.core.BackgroundResource;
 import com.google.common.base.Preconditions;
 import com.google.common.collect.ImmutableList;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
 import java.io.IOException;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 import java.util.concurrent.TimeUnit;
 import javax.annotation.Nullable;
 
 /** Implementation of HttpJsonChannel which can issue http-json calls. */
 @BetaApi
 public class ManagedHttpJsonChannel implements HttpJsonChannel, BackgroundResource {
   private static final JsonFactory JSON_FACTORY = GsonFactory.getDefaultInstance();
+  private static final ExecutorService DEFAULT_EXECUTOR =
```

Comment:
We could do something similar for grpc channel provider. I thought it might be better to use the default grpc one in case grpc update it to something more efficient in the future. And also since grpc is already providing default executor, we don't need to repeat it in gax? 

I'm setting a default http executor here because I'm deprecating needsExecutor, and http channel provider doesn't have a default one. If we're not sure about setting it to `Executors.newCachedThreadPool()`, maybe we can use the default [gax executor](https://github.com/googleapis/gax-java/blob/master/gax/src/main/java/com/google/api/gax/core/InstantiatingExecutorProvider.java#L63) here? And I think it's better to separate channel executor and background executor?